python/lsst/daf/butler/script/removeRuns.py

```diff
@@ -96,17 +95,20 @@ def _getCollectionInfo(
         )
     except MissingCollectionError:
         collectionNames = []
+
+    dataset_types = butler.registry.queryDatasetTypes(...)
     runs = []
     datasets: dict[str, int] = defaultdict(int)
     for collectionName in collectionNames:
         assert butler.registry.getCollectionType(collectionName).name == "RUN"
         parents = butler.registry.getCollectionParentChains(collectionName)
         runs.append(RemoveRun(collectionName, list(parents)))
-        all_results = butler.registry.queryDatasets(..., collections=collectionName)
-        assert isinstance(all_results, DatasetQueryResults)
-        for r in all_results.byParentDatasetType():
-            if r.any(exact=False, execute=False):
-                datasets[r.parentDatasetType.name] += r.count(exact=False)
+        with butler._query() as query:
+            for dt in dataset_types:
+                results = query.datasets(dt, collections=collectionName)
+                if results.any(exact=False, execute=False):
+                    datasets[dt.name] += results.count(exact=False)
```

Comment:
It'll be slightly faster to just execute `count` and ignore the result if it is zero.

Reply:
Thanks. That crossed my mind. I was wondering whether the `execute=False` on the `any` vs the `execute=True` on the `count` would be a performance difference but I didn't run any tests.

Reply:
The `any(execute=False)` summary check is going to be run internally by `count` (or any other query execution method) automatically, so in this case it's just doing that check twice.  It should be a fast check the second time (once the cache is populated), so it's not a big deal, but it is unnecessary.